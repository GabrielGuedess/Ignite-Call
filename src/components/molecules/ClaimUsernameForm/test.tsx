import { render } from '@testing-library/react';

import { ClaimUsernameForm } from '.';

// eslint-disable-next-line @typescript-eslint/no-var-requires
const useRouter = jest.spyOn(require('next/router'), 'useRouter');

useRouter.mockImplementation(() => ({
  push: jest.fn(),
}));

describe('<ClaimUsernameForm />', () => {
  it('should render the claimUsernameForm correctly', () => {
    const { container } = render(<ClaimUsernameForm />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
