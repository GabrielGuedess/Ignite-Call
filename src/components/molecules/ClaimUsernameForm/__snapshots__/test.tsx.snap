// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClaimUsernameForm /> should render the claimUsernameForm correctly 1`] = `
<form
  class="c-cAYWOa c-dWDLra"
>
  <div
    class="c-jlrPyi c-jlrPyi-cAYvGz-size-sm"
  >
    <span
      class="c-bOKaMi"
    >
      ignite.com/
    </span>
    <input
      class="c-grKefI"
      name="username"
      placeholder="seu-usuÃ¡rio"
    />
  </div>
  <button
    class="c-jZXQKc c-jZXQKc-hazApa-variant-primary c-jZXQKc-eeFPgf-size-sm"
    type="submit"
  >
    Reservar
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 256 256"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        fill="none"
        height="256"
        width="256"
      />
      <line
        fill="none"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="16"
        x1="40"
        x2="216"
        y1="128"
        y2="128"
      />
      <polyline
        fill="none"
        points="144 56 216 128 144 200"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="16"
      />
    </svg>
  </button>
</form>
`;
